examples/tutorials/asr_inference_with_ctc_decoder_tutorial.py

```diff
@@ -222,28 +228,215 @@ def forward(self, emission: torch.Tensor) -> str:
+# please refer to the
+# :py:func:`documentation <torchaudio.prototype.ctc_decoder.kenlm_lexicon_decoder>`.  # noqa
+#
 
 
 ######################################################################
-# We see that the transcript with the lexicon-constrained beam search
-# decoder consists of real words, while the greedy decoder can predict
-# incorrectly spelled words like “hundrad”.
+# Helper Function
+# ~~~~~~~~~~~~~~~
+#
+
+
+def get_transcript_and_time(decoder, emission):
+    start_time = time.monotonic()
+    result = decoder(emission)
+    decode_time = time.monotonic() - start_time
+
+    transcript = " ".join(result[0][0].words).strip()
+    return transcript, decode_time
+
+
+######################################################################
+# nbest
+# ~~~~~
+#
+# This parameter indicates the number of best Hypothesis to return, which
+# is a property that is not possible with the greedy decoder. For
+# instance, by setting ``nbest=3`` when constructing the beam search
+# decoder earlier, we can now access the hypotheses with the top 3 scores.
+#
+
+for i in range(3):
+    transcript = " ".join(beam_search_result[0][i].words).strip()
+    score = beam_search_result[0][i].score
+    print(f"{transcript} (score: {score})")
+
+
+######################################################################
+# beam size
+# ~~~~~~~~~
+#
+# The ``beam_size`` parameter determines the maximum number of best
+# hypotheses to hold after each decoding step. Using larger beam sizes
+# allows for exploring a larger range of possible hypotheses which can
+# produce hypotheses with higher scores, but it is computationally more
+# expensive and does not provide additional gains beyond a certain point.
+#
+# In the example below, we see improvement in decoding quality as we
+# increase beam size from 1 to 5 to 50, but notice how using a beam size
+# of 500 provides the same output as beam size 50 while increase the
+# computation time.
+#
+
+beam_sizes = [1, 5, 50, 500]
+
+for beam_size in beam_sizes:
+    beam_search_decoder = kenlm_lexicon_decoder(
+        lexicon=lexicon_file,
+        tokens=tokens,
+        kenlm=kenlm_file,
+        beam_size=beam_size,
+        lm_weight=LM_WEIGHT,
+        word_score=WORD_SCORE,
+    )
+
+    transcript, decode_time = get_transcript_and_time(beam_search_decoder, emission)
+    print(f"beam size {beam_size:<3}: {transcript} ({decode_time:.4f} secs)")
+
+
+######################################################################
+# beam size token
+# ~~~~~~~~~~~~~~~
+#
+# The ``beam_size_token`` parameter corresponds to the number of tokens to
+# consider for expanding each hypothesis at the decoding step. Exploring a
+# larger number of next possible tokens increases the range of potential
+# hypotheses at the cost of computation. This parameter is especially
+# helpful for lexicon-free decoders, where the number of possible next
+# tokens is not already partially constrained by the lexicon.
```

Comment:
>  This parameter is especially helpful for lexicon-free decoders, 
> where the number of possible next tokens is not already
> partially constrained by the lexicon.

Although this is informative and educative, we do not support a lexicon-free decoder (yet?), so maybe not as useful to users?